connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -539,6 +539,15 @@ class SparkConnectPlanner(session: SparkSession) {
    * @return
    */
   private def transformScalarFunction(fun: proto.Expression.UnresolvedFunction): Expression = {
+    if (fun.getFunctionName == "product") {
```

Comment:
Had a discussion with @cloud-fan , we don't want to expose all expressions in SQL syntax, and `product` is such an example.

I think we will also deal with dedicated expressions for Pandas-on-Spark in this way.

Reply:
I was thinking we wanted to add `product` to `FunctionRegistry` as a follow-up for https://github.com/apache/spark/pull/38905. I didn't see how we've done this in this PR. Would you help me understand?

Reply:
What is the reason for not allowing this in SQL? And defining it in the function registry should not necessarily mean it can be used from SQL. It might be easier to disallow usage in SQL in the function registry then dealing with it here. I think the function registry is the right place for this.

Reply:
cc @cloud-fan

Reply:
We don't have a mechanism to disallow functions in `FunctionRegistry`. We can build such a mechanism but it's simpler to define the list of "hidden functions" here.

There is no specific reason why `product` can't be a SQL function. We just need a separate PR to propose it and justify it (is it in SQL standard? is it common supported in other databases?)